### PR TITLE
Add support for wildcard (`some*match`) uninstalls

### LINKED ISSUE #434: libmamba solver doesn't match when uninstalling with wildcard
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

When attempting `conda uninstall thing*thing`, i.e. uninstalling all packages matching a wildcard, the `libmamba` solver never matches anything. Consider the following:

```console
$ conda uninstall "lib*blas" --solver libmamba

PackagesNotFoundError: The following packages are missing from the target environment:
  - lib*blas

```

Specifying the `classic` solver resolves the right set of packages to remove:

```console
$ conda uninstall "lib*blas" --solver classic
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/duncan/opt/mambaforge/envs/test

  removed specs:
    - lib*blas


The following packages will be REMOVED:

  _libgcc_mutex-0.1-conda_forge
  _openmp_mutex-4.5-2_kmp_llvm
  blas-2.121-openblas
  blas-devel-3.9.0-21_linux64_openblas
  libblas-3.9.0-21_linux64_openblas
  libcblas-3.9.0-21_linux64_openblas
  libgcc-ng-13.2.0-h807b86a_4
  libgfortran-ng-13.2.0-h69a702a_4
  libgfortran5-13.2.0-ha4646dd_4
  liblapack-3.9.0-21_linux64_openblas
  liblapacke-3.9.0-21_linux64_openblas
  libopenblas-0.3.26-pthreads_h413a1c8_0
  libstdcxx-ng-13.2.0-h7e041cc_4
  libzlib-1.2.13-hd590300_5
  llvm-openmp-17.0.6-h4dfa4b3_0
  openblas-0.3.26-pthreads_h7a3da1a_0
  zstd-1.5.5-hfc55251_0


Proceed ([y]/n)? n


CondaSystemExit: Exiting.
```

### Conda Info

```shell
$ conda info

     active environment : test
    active env location : /home/duncan/opt/mambaforge/envs/test
            shell level : 2
       user config file : /home/duncan/.condarc
 populated config files : /home/duncan/opt/mambaforge/.condarc
                          /home/duncan/.condarc
          conda version : 23.11.0
    conda-build version : 3.28.4
         python version : 3.11.6.final.0
                 solver : libmamba (default)
       virtual packages : __archspec=1=skylake
                          __conda=23.11.0=0
                          __glibc=2.36=0
                          __linux=5.15.133.1=0
                          __unix=0=0
       base environment : /home/duncan/opt/mambaforge  (writable)
      conda av data dir : /home/duncan/opt/mambaforge/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/igwn/linux-64
                          https://conda.anaconda.org/igwn/noarch
          package cache : /home/duncan/opt/mambaforge/pkgs
                          /home/duncan/.conda/pkgs
       envs directories : /home/duncan/opt/mambaforge/envs
                          /home/duncan/.conda/envs
               platform : linux-64
             user-agent : conda/23.11.0 requests/2.31.0 CPython/3.11.6 Linux/5.15.133.1-microsoft-standard-WSL2 debian/12.4 glibc/2.36 solver/libmamba conda-libmamba-solver/24.1.0 libmambapy/1.5.6
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```

### Conda Config

```shell
==> /home/duncan/opt/mambaforge/.condarc <==
channels:
  - conda-forge

==> /home/duncan/.condarc <==
channel_priority: strict
channels:
  - conda-forge
  - igwn
solver: libmamba
conda_build:
  error_overdepending: True
  error_overlinking: True
```

### Conda list

```shell
$ conda list --show-channel-urls
# packages in environment at /home/duncan/opt/mambaforge/envs/test:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                  2_kmp_llvm    conda-forge
blas                      2.121                  openblas    conda-forge
blas-devel                3.9.0           21_linux64_openblas    conda-forge
libblas                   3.9.0           21_linux64_openblas    conda-forge
libcblas                  3.9.0           21_linux64_openblas    conda-forge
libgcc-ng                 13.2.0               h807b86a_4    conda-forge
libgfortran-ng            13.2.0               h69a702a_4    conda-forge
libgfortran5              13.2.0               ha4646dd_4    conda-forge
liblapack                 3.9.0           21_linux64_openblas    conda-forge
liblapacke                3.9.0           21_linux64_openblas    conda-forge
libopenblas               0.3.26          pthreads_h413a1c8_0    conda-forge
libstdcxx-ng              13.2.0               h7e041cc_4    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
llvm-openmp               17.0.6               h4dfa4b3_0    conda-forge
openblas                  0.3.26          pthreads_h7a3da1a_0    conda-forge
zstd                      1.5.5                hfc55251_0    conda-forge
```

### Additional Context

`test` environment created using `conda create -n test blas`.

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Closes #434 

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [X] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Comment by jaimergp** [comment]
Test reproduces error. Pushing fix.

**Comment by jaimergp** [comment]
This is ready for review @jezdez, but I wonder if we should have this test added to `conda/conda` instead?

**Comment by jaimergp** [comment]
@zklaus - would you like trying to upstream the test added in this PR to `conda/conda`? 🙏

**Comment by zklaus** [comment]
Will do ;)

**Comment by jaimergp** [comment]
Passing now:

```
tests/cli/test_main_remove.py::test_remove_globbed_package_names[libmamba] PASSED [ 84%]
```

Test is included in Upstream / Integration / Group 1 (for libmamba), and Group 3 (for classic).

Ready for review!

File: conda_libmamba_solver/state.py [hunk 1]
```diff
@@ -544,21 +544,28 @@ def early_exit(self) -> Dict[str, PackageRecord]:
...
+            # When 'remove --force' is set, remove the package without solving.
+            if sis.force_remove:
+                force_remove_solution = self.current_solution
+            not_installed = []
+            for name, spec in sis.requested.items():
+                for record in sis.installed.values():
+                    if spec.match(record):
+                        if sis.force_remove:
+                            force_remove_solution.remove(record)
+                        break
```
Comment (jezdez) [inline conda_libmamba_solver/state.py:558]: Remind me, will this break out of both for loops?

**Review by jezdez — COMMENTED** [review]
Any chance to add a test for the new feature?

**Comment by jaimergp** [comment]
@jezdez The test was here initially, then Klaus upstreamed it at https://github.com/conda/conda/pull/13579 because it's a conda feature, not a CLS one. That's what I meant with my earlier comment.

File: conda_libmamba_solver/state.py (hunk 1 above)
Comment (jaimergp) [reply conda_libmamba_solver/state.py:558]: Only the innermost one, which is what I intend here. We need to check all requested specs, and see if they are indeed installed. If they are not, then we add it to the list to report the error later.

**Review by jezdez — APPROVED** [review]

**Pull request merged**